File: exercism/clojure/sum-of-multiples/src/sum_of_multiples.clj
(ns sum-of-multiples)

(defn compute-modulos [seq-of-multiples number]
  (map (partial mod number) seq-of-multiples))

(defn has-some-multiples? [seq-of-multiples number]
  (some (partial = 0) (compute-modulos seq-of-multiples number)))

(defn sum-of-multiples [seq-of-multiples number]
  (->>
   (range 1 number)
   (filter (partial sum-of-multiples/has-some-multiples? seq-of-multiples))
   (distinct)
   (reduce +)))

File: exercism/java/sum-of-multiples/src/main/java/SumOfMultiples.java
import java.util.stream.IntStream;

class SumOfMultiples {

    private int number;
    private int[] multiples;

    SumOfMultiples(int number, int[] set) {
        this.number = number;
        this.multiples = set;
    }

    int getSum() {
        if (number < 2) {
            return 0;
        }
        return IntStream.range(1, this.number).filter(i -> IntStream.of(this.multiples).filter(j -> j != 0).anyMatch(j -> i % j == 0)).sum();
    }

}
